pyomo/environ/tests/test_environ.py

```diff
@@ -151,6 +152,8 @@ def test_tpl_import_time(self):
         ref.add('pyutilib')
         if numpy_available:
             ref.add('numpy')
+        if cyipopt_available:
+            ref.add('cyipopt')
```

Comment:
We don't want to add cyipopt to this list: this is indicating that cyipopt is being unconditionally imported by `pyomo.environ`.  We should re-think the imports so that we don't have to update this list.

Reply:
Is this because `PyomoCyIpoptSolver` is registered as a plugin so it can be accessed via `pyo.SolverFactory("cyipopt")`?

Reply:
Yes: `pyomo.environ` imports `pyomo.contrib.pynumero.plugins` which imports `pyomo.contrib.pynumero.algorithms.solers.cyipopt_solver`

Reply:
I tried using a deferred import (via `attempt_import) for `pyomo.contrib.pynumero.algorithms.solvers.cyipopt_solver` in `pynumero.plugins`, but that didn't change anything, nor did moving `from .algorithms.solvers.cyipopt_solver import PyomoCyIpoptSolver` to within the `load` function of `pynumero.plugins`.

Do you have any ideas?